### PR TITLE
Override gate.controlled() for GlobalPhaseGate to return a ZPowGate

### PR DESCRIPTION
**Pull request by tanujkhattar**
`ZPowGate`, `CZPowGate` and `CCZPowGate` currently have specialized `Gate.controlled()` overrides such that `cirq.Z.controlled()` returns a `cirq.CZ` instead of a `cirq.ControlledGate(sub_gate=cirq.Z)`. This is nice and useful for writing controlled versions of common gates. 

This PR extends the same idea to `GlobalPhaseGate`, because a controlled `GlobalPhaseGate` is equivalent to applying a `cirq.ZPowGate`.

### CODE REVIEW

File: cirq-core/cirq/ops/global_phase_op.py [hunk 1]
```diff
@@ -91,6 +91,29 @@ def _resolve_parameters_(
...
+    def controlled(
+        self,
+        num_controls: Optional[int] = None,
+        control_values: Optional[
+            Union[cv.AbstractControlValues, Sequence[Union[int, Collection[int]]]]
+        ] = None,
+        control_qid_shape: Optional[Tuple[int, ...]] = None,
+    ) -> raw_types.Gate:
+        result = super().controlled(num_controls, control_values, control_qid_shape)
+        if (
```
Comment (dstrain115) [inline cirq-core/cirq/ops/global_phase_op.py:103]: Maybe a line comment explaining this condition (can probably use something similar to the PR description.

File: cirq-core/cirq/ops/global_phase_op_test.py [hunk 2]
```diff
@@ -279,3 +279,20 @@ def test_resolve_error(resolve_fn):
     gpt = cirq.GlobalPhaseGate(coefficient=t)
     with pytest.raises(ValueError, match='Coefficient is not unitary'):
         resolve_fn(gpt, {'t': -2})
+
+
+@pytest.mark.parametrize(
```
Comment (dstrain115) [inline cirq-core/cirq/ops/global_phase_op_test.py:284]: Optional:  should we add a test for some of the other conditions in the code, like the ProductOfSums, that it does not return the target gate?

**Review by dstrain115 — APPROVED** [review]

File: cirq-core/cirq/ops/global_phase_op.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/global_phase_op.py:103]: done

File: cirq-core/cirq/ops/global_phase_op_test.py (hunk 2 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/global_phase_op_test.py:284]: done

**Pull request merged**